plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeProjectionPushdownPlans.java

```diff
@@ -0,0 +1,261 @@
+    public void cleanup()
+            throws Exception
+    {
+        if (baseDir != null) {
+            deleteRecursively(baseDir.toPath(), ALLOW_INSECURE);
+        }
+    }
+
+    @Test
+    public void testPushdownDisabled()
+    {
+        String testTable = "test_disabled_pushdown" + randomNameSuffix();
+
+        Session session = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalogSessionProperty(CATALOG, "projection_pushdown_enabled", "false")
+                .build();
+
+        getQueryRunner().execute(format(
+                "CREATE TABLE %s (col0) AS SELECT CAST(row(5, 6) AS row(a bigint, b bigint)) AS col0 WHERE false",
+                testTable));
+
+        assertPlan(
+                format("SELECT col0.a expr_a, col0.b expr_b FROM %s", testTable),
+                session,
+                any(
+                        project(
+                                ImmutableMap.of("expr", expression("col0[1]"), "expr_2", expression("col0[2]")),
+                                tableScan(testTable, ImmutableMap.of("col0", "col0")))));
+    }
+
+    @Test
+    public void testDereferencePushdown()
+    {
+        String testTable = "test_simple_projection_pushdown" + randomNameSuffix();
+        QualifiedObjectName completeTableName = new QualifiedObjectName(CATALOG, SCHEMA, testTable);
+
+        getQueryRunner().execute(format(
+                "CREATE TABLE %s (col0, col1) WITH (partitioned_by = ARRAY['col1']) AS" +
+                        " SELECT CAST(row(5, 6) AS row(x bigint, y bigint)) AS col0, 5 AS col1",
+                testTable));
+
+        Session session = getQueryRunner().getDefaultSession();
+
+        Optional<TableHandle> tableHandle = getTableHandle(session, completeTableName);
+        assertThat(tableHandle).as("expected the table handle to be present").isPresent();
+
+        Map<String, ColumnHandle> columns = getColumnHandles(session, completeTableName);
+
+        DeltaLakeColumnHandle column0Handle = createColumnHandleWithIndex((DeltaLakeColumnHandle) columns.get("col0"), 0);
+        DeltaLakeColumnHandle column1Handle = createColumnHandleWithIndex((DeltaLakeColumnHandle) columns.get("col1"), 1);
+
+        DeltaLakeColumnHandle columnX = createProjectedColumnHandle(column0Handle, 0, ImmutableList.of(0), ImmutableList.of("x"));
+        DeltaLakeColumnHandle columnY = createProjectedColumnHandle(column0Handle, 0, ImmutableList.of(1), ImmutableList.of("y"));
+
+        // Simple Projection pushdown
+        assertPlan(
+                "SELECT col0.x expr_x, col0.y expr_y FROM " + testTable,
+                any(tableScan(
+                        equalTo(((DeltaLakeTableHandle) tableHandle.get().getConnectorHandle()).withProjectedColumns(Set.of(columnX, columnY))),
+                        TupleDomain.all(),
+                        ImmutableMap.of("col0.x", equalTo(columnX), "col0.y", equalTo(columnY)))));
+
+        // Projection and predicate pushdown
```

Comment:
Before your change did file pruning during splits generation on coordinator due to predicates on dereferenced struct fields work ? If not, might want to see if it works now and add a test.
Similar question about column statistics for the CBO on dereferenced struct fields.

Reply:
File pruning neither works before the change nor after the change. It needs to be fixed.

I have raised an issue for the same https://github.com/trinodb/trino/issues/17159